[en] Add term: 'Event-Driven Architecture' (#637)
cncf/glossary: content/en/event-driven_architecture.md

---
title: Event-Driven Architecture
status: Feedback Appreciated
category: concept
---
## What it is

> Event-driven architecture is a software architecture that promotes the creation, processing, and consumption of events. An event can be seen as any change of state or other significant occurrence in hardware or software. This architecture creates the structure in which events can be properly routed from their source to the desired receivers.

Author: I went ahead and made a change to the description. Let me know what you think!

## Problem it addresses

> As more data is becoming real time in nature, it becomes difficult to find reliable ways to ensure that events are captured then routed to the service that needs to process the event request. There is also the need to guarantee delivery of the event to the proper service.

Reviewer: ```suggestion
As more data becomes real-time, finding reliable ways to ensure that events are captured and routed to the service that must process the event request gets increasingly challenging. There is also the need to guarantee delivery of the event to the proper service.
```

Reviewer: This seems rather vague: "There is also the need to guarantee delivery of the event to the proper service." Maybe something like "Some applications may even need an event delivery guarantee for specific services to function as designed." >> this may not be correct but I feel that "there is also a need" needs more context if that makes sense.

Author: I just pushed some code changes

> As more data is becomes real-time, finding reliable ways to ensure that events are captured amd routed to the service that must process event request gets increasingly challenging. Many applications need a way to guarantee event delivery to ensure that the application functions designed. With multiple event sources and event receiver, it also becomes challenging to ensure that events are properly routed to their intended recipient services.

Reviewer: This is marked as resolved but the typos are still there

## How it helps

> Event-driven architecture establishes the producer (source) of an event and the consumer (receiver). This architecture ensures that services remain decoupled and that events are properly routed from the producer to the consumer. The producer will take the incoming event, usually by HTTP protocol and then route the event information.

Reviewer: ```suggestion
Event-driven architectures establish event producers (source) and consumers (receiver). This architecture ensures that services remain decoupled and that events are properly routed from the producer to the consumer. The producer will take the incoming event, usually by HTTP protocol, and then route the event information.
```

Author: Made the requested adjustments :-)

> Event-driven architecture establishes the event producers (source) consumers (receiver). This architecture ensures that services remain decoupled and that events are properly routed from the producer to the consumer. The producer will take the incoming event, usually by HTTP protocol and then route the event information.

Reviewer: Same here. This is marked as resolved but the typos are still here. It should be either:

- An event-driven architecture establishes OR
- Event-driven architectures establish 

An "and" appears to be missing and so is a comma. If you disagree, could you please explain why? Otherwise, we'll be running in circles. I'll keep correcting the same things, and you'll keep resolving without committing :)

---
title: Event-Driven Architecture
status: Feedback Appreciated
category: concept
---
## What it is

> Event-driven architecture is a software architecture that promotes the creation, processing, and consumption of events. An event is any change of state or other significant occurrence in hardware or software. This architecture creates the structure in which events can be properly routed from their source to the desired receivers.

Reviewer: Not seeing any chance here. Maybe you didn't commit?

Author: You are right, I did not commit before push. It's Friday

---
title: Event-Driven Architecture
status: Feedback Appreciated
category: concept
---

## What it is

> It is a software architecture that promotes the creation, processing, and consumption of events. An event can be seen as any change of state or other significant occurance in hardware or software. This architecture creates the structure in which events can be properly routed from their source to the desired receivers.

Reviewer 2: If 'occurance' is typo of 'occurrence', then please fix it. 😊

Author: That file is outdated, I updated a recent one

> It is a software architecture that promotes the creation, processing, and consumption of events. An event can be seen as any change of state or other significant occurence in hardware or software. This architecture creates the structure in which events can be properly routed from their source to the desired receivers.

Reviewer 2: Spellchecker found this 😊

```suggestion
It is a software architecture that promotes the creation, processing, and consumption of events. An event can be seen as any change of state or other significant occurrence in hardware or software. This architecture creates the structure in which events can be properly routed from their source to the desired receivers.
```

---
title: Event-Driven Architecture
status: Feedback Appreciated
category: concept
---
## What it is
Event-driven architecture is a software architecture that promotes the creation, processing, and consumption of events. 
An event is any change to an application's state. 
For example, hailing a ride on a ridesharing app represents an event. 

> This architecture creates the structure in which events can be properly routed from their source to the desired receivers.

Reviewer: ```suggestion
This architecture creates the structure in which events can be properly routed from their source (my app requesting that ride) to the desired receivers (the apps of available drivers nearby).
```

## Problem it addresses
As more data becomes real-time, finding reliable ways to ensure that events are captured and routed to the appropriate [service](https://glossary.cncf.io/service/) that must process event requests gets increasingly challenging. 
Many applications need a way to guarantee event delivery to ensure the application functions as designed.
With multiple event sources and event receivers, it also becomes challenging to ensure that events are properly routed to their intended recipient services.

> ## How it helps

Reviewer 3: How it helps is still a little on the jargon-y side. I think talk of producers and consumers should either be omitted or referred to in the What it is section. IMO event driven architecture helps by allowing organizations/developers to focus on reacting to events and save on compute costs by reducing the number of app instances that need to exist during low traffic periods. 

I think the entry is in great shape overall and I'm happy to set up a pairing session to finish it if that would be helpful. Thanks a ton for your work here.

Author: Made some changes

> title: Event-Driven Architecture

Reviewer 4: Please change the file name to content/en/**event-driven-architecture**.md

- The guide for file name (in the recent https://glossary.cncf.io/contribute/)

> Add the **term name** in the URL (no capitalization, no spaces, no parentheses, and **use hyphens to separate words**) and **.md** at the end (note: if your preview doesn’t work, you probably forgot to add .md at the end). Now paste the template content below. Copy and paste your definition into the file. To make reviews easier, please use [semantic line breaks](https://sembr.org/) (e.g. one line per sentence). Note that GitHub uses markdown to format the text (e.g., hyperlink, bold, italic). Please refer to this [markdown cheat sheet](https://www.markdownguide.org/cheat-sheet/). To verify you’ve used markdown as intended, go to “Preview”.

> title: Event-Driven Architecture

Author: Done
